website/docs/reference/global-configs/warnings.md

```diff
@@ -2,129 +2,216 @@
+:::
+
+Instead of using the `--warn-error` flag to promote _all_ warnings to errors, you can use [`--warn-error-options`](#use---warn-error-options-for-targeted-warnings) flag to promote _specific_ warnings to errors, including:
+
+- Test warnings with the `--warn-error-options '{"include": ["LogTestResults"]}'` flag.
+- Jinja-level warnings with the `--warn-error-options '{"include": ["JinjaLogWarning"]}'` flag or [`exceptions.warn`](/reference/dbt-jinja-functions/exceptions#warn).
+- Selection issues with the `--warn-error-options '{"include": ["NoNodesForSelectionCriteria"]}'` flag.
+- Adapter deprecation warnings with the `--warn-error-options include": ["AdapterDeprecationWarning"]}'` flag.
+
+</VersionBlock>
 
 <File name='Usage'>
 
-```text
-dbt --warn-error run
-...
-```
+  ```text
+  dbt --warn-error run
+  ```
 
 </File>
 
-### Using `--warn-error-options` for targeted warnings
+## Use `--warn-error-options` for targeted warnings
 
-Converting any warnings to errors may suit your needs perfectly, but there may be some warnings you just don't care about, and some you care about a lot. The `WARN_ERROR_OPTIONS` config or `--warn-error-options` flag gives you more granular control over _exactly which types of warnings_ are treated as errors. 
+In some cases, you may want to convert _all_ warnings to errors. However, when you want _some_ warnings to stay as warnings and only promote or silence specific warnings you can instead use `--warn-error-options`. The `WARN_ERROR_OPTIONS` config or `--warn-error-options` flag gives you more granular control over _exactly which types of warnings_ are treated as errors. 
 
-:::info `WARN_ERROR` and `WARN_ERROR_OPTIONS` configs are mutually exclusive
-`WARN_ERROR` and `WARN_ERROR_OPTIONS` configs are mutually exclusive. You can only specify one, even when you're specifying the config in multiple places (e.g. env var + CLI flag), otherwise, you'll see a usage error.
+:::info `WARN_ERROR` and `WARN_ERROR_OPTIONS` are mutually exclusive
+`WARN_ERROR` and `WARN_ERROR_OPTIONS` are mutually exclusive. You can only specify one, even when you're specifying the config in multiple places (like env var or a flag), otherwise, you'll see a usage error.
 :::
 
+<VersionBlock lastVersion="1.7">
+
+- Warnings that should be treated as errors can be specified through `include` and/or `exclude` parameters. Warning names can be found in:
+  - [dbt-core's types.py file](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/types.py), where each class name that inherits from `WarnLevel` corresponds to a warning name (e.g. `AdapterDeprecationWarning`, `NoNodesForSelectionCriteria`) 
+  - Downloading the JSON output logs from a run and searching for the warning.
```

Comment:
you can do this using `--log-format json` flag

Reply:
oh that's much easier! when i was testing all of this,  i had to download the log and search for it!

<img width="457" alt="Screenshot 2025-04-08 at 20 17 25" src="https://github.com/user-attachments/assets/4d0a67ae-1db3-4515-b0cb-53cb159f3ef6" />

Reply:
oof you went for hard mode 😆